libraries/script-engine/src/WebSocketClass.cpp

```diff
@@ -97,6 +103,25 @@ void WebSocketClass::handleOnMessage(const QString& message) {
     }
 }
 
+void WebSocketClass::handleOnBinaryMessage(const QByteArray& message) {
+    if (_onMessageEvent.isFunction()) {
+        QScriptValueList args;
+        QScriptValue arg = _engine->newObject();
+        QScriptValue data = _engine->newVariant(QVariant::fromValue(message));
+        QScriptValue ctor = _engine->globalObject().property("ArrayBuffer");
+        ArrayBufferClass* array = qscriptvalue_cast<ArrayBufferClass*>(ctor.data());
+        QScriptValue arrayBuffer;
+        if (!array) {
+            qWarning() << "WebSocketClass::handleOnBinaryMessage !ArrayBuffer";
```

Comment:
Would you still want to send an event when there is no data present? or could we return here?

Reply:
I'd say still send the event: Interface has received an event from the socket so we should pass that info on to the script, even if there is not data.

Reply:
👍